### PR TITLE
Some Doc polishing

### PR DESCRIPTION
**Pull request by hynek**
This mostly applies changes I’ve mentioned on the mailing list and makes the README a bit more like the cryptography one.

I’ve also double checked all links, fixed broken ones and boy-scouted a section that contained a broken link.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -17,16 +17,27 @@ pyOpenSSL -- A Python wrapper around the OpenSSL library
...
-* SSL.Connection objects, wrapping the methods of Python's portable sockets
+* ``SSL.Connection`` objects, wrapping the methods of Python's portable sockets
 * Callbacks written in Python
 * Extensive error-handling mechanism, mirroring OpenSSL's error codes
 
 ... and much more.
 
-See the file INSTALL.rst for installation instructions.
+You can find more information in the documentation_.
+Development takes place on GitHub_.
```
Comment (Lukasa) [inline README.rst]: Missing backticks around these links.

File: README.rst [hunk 2]
```diff
@@ -17,16 +17,27 @@ pyOpenSSL -- A Python wrapper around the OpenSSL library
...
-See https://mail.python.org/mailman/listinfo/pyopenssl-users for the discussion mailing list.
+If you run into bugs, you can file them in our `issue tracker`_.
+
+We maintain a cryptography-dev_ mailing list for both user and development discussions.
+
+You can also join ``#cryptography-dev`` on Freenode to ask questions or get involved.
+
+
+.. _documentation: https://pyopenssl.readthedocs.org/
+.. _`issue tracker`: https://github.com/pyca/pyopenssl/issues
```
Comment (Lukasa) [inline README.rst]: Superfluous backticks here.

**Comment by Lukasa** [comment]
My notes about backticks are purely advisory (I find it weird to not consistently use backticks), so I'm going to merge this anyway: @hynek you can change that if you care (I suspect you don't).

File: README.rst (hunk 1 above)
Comment (hynek) [inline README.rst]: Nah, they work fine?

**Pull request merged**